public/js/surveyBuilderForm.js

```diff
@@ -190,4 +154,72 @@ $(function() {
         }
     }
 
-})
\ No newline at end of file
+/**
+ * returns a string of the response input options based on the question type
+ *
+ * @param type STRING type of response
+ * @param options ARRAY response options from the DOM
+ *
+ * @returns STRING html input(s) for question response
+ */
+function getNewQuestionResponse(type, options) {
+
+    var response = '<div class="options">'
+
+    if(type == 'text') {
+        response += '<input type="text" disabled >'
+    } else {
+        options.each(function(key, option)
+        {
+            if($(this).prop('id') != 'option-text'){
+                response += '<div><input type="' + type + '" disabled value="' + option.value + '"> ' + option.value + '</div>'
+            }
+        })
+    }
+
+    response += '</div>'
+
+    return response
+}
+
+/**
+ * creates question preview div using passed information and appends to container
+ *
+ * @param $container JQUERYSELECTOR container to append new question to
+ * @param question STRING question text
+ * @param response STRING html input(s) for question response
+ * @param $required JQUERYSELECTOR required checkbox
+ */
```

Comment:
change the data type from JQUERYSELECTOR to OBJECT, as it is the selector's return value that we are assigning to the varibable, not the selector itself. jQuery selectors return jQuery objects, which are themselves JS objects.